### PR TITLE
update lesson for PostgreSQLwith separate classes to persist and retr…

### PR DESCRIPTION
**Pull request by linda-seiter**
…ieve an entity

### CODE REVIEW

**Review by developerjaym — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -145,107 +125,128 @@ public class Student {
...
+
 ```
 
-The `@Entity` annotation tells program that JPA has to manage this class and
-will be using this to communicate with the database. The `@Table` annotation is
-being used to set the table name in the database. Finally, the `@Id` annotation
-specifies the property that will be used as the unique identifier for rows in
-the database.
+
+- The `@Entity` annotation tells program that JPA has to manage this class and
```
Comment (kcleland0818) [inline README.md]: Typo: "The `@Entity` annotation tells program that JPA has to manage this class..."; add "the" between "tells" and "program".

File: README.md [hunk 2]
```diff
@@ -145,107 +125,128 @@ public class Student {
...
+  will be using this to communicate with the database.
+- The `@Table` annotation sets the table name in the database.  The class name is used if this is omitted.
+- The `@Id` annotation specifies the property that will be used
+  as the unique identifier (i.e. primary key) for rows in the database.
+
+A class with the `@Entity` annotation should have a no-args constructor.
+Java generates a default no-args constructor for `Student` since a constructor
+is not explicitly defined.  The entity class must have a primary key that uniquely identifies it.
+It is also recommended to include setter methods.
```
Comment (kcleland0818) [inline README.md]: And getter methods too?

File: README.md [hunk 3]
```diff
@@ -145,107 +125,128 @@ public class Student {
...
+1. Run the `JpaCreateStudent.main` method. This will create a `STUDENT_DATA`
+   table in the PostgreSQL `student_db` database and add a new row. 
+2. In IntelliJ, check out the “Run” tab to see the exact query that Hibernate used
+   to create the table and insert the data. It shows the SQL queries because we enabled this behavior in
+   the `persistence.xml` file.
+
+```text
+Hibernate: 
+    
     create table STUDENT_DATA (
```
Comment (kcleland0818) [inline README.md:224]: When I look at the table that is created in pgAdmin, it looks like the table name is all lowercase. Does casing not matter then in Postgres?

File: README.md (hunk 1 above)
Comment (linda-seiter) [reply README.md]: fixed

File: README.md (hunk 2 above)
Comment (linda-seiter) [reply README.md]: fixed

File: README.md (hunk 3 above)
Comment (linda-seiter) [reply README.md:224]: both queries seem to work in postgres:

select * from STUDENT_DATA;
select * from student_data;

**Review by kcleland0818 — APPROVED** [review]

**Review by developerjaym — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -31,7 +31,7 @@ database. These configurations are added to the `persistence.xml` file.
 2. Create a file named `persistence.xml` in the `META-INF` folder.
 3. Your project directory structure should look like the following:  
 
-![project structure persistence.xml](https://curriculum-content.s3.amazonaws.com/6036/java-mod-5-jpa/persistence_xml.png)  
+![persistence.xml](https://curriculum-content.s3.amazonaws.com/6036/java-mod-5-jpa/persistence_xml_project.png)
```
Comment (kcleland0818) [inline README.md]: These two images look the same? Was that on purpose??

Comment (linda-seiter) [reply README.md]: intellij sometimes won't refresh the image in the preview, so renaming forces it.

I updated the images to crop out the project path.

Comment (kcleland0818) [reply README.md]: If you don't want to rename it, you can try this trick that Liz sent to me a while ago:

curl -X PURGE {url of cached image}

Source: [from Liz](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/about-anonymized-urls#removing-an-image-from-camos-cache). "That caching issue is a real pita. It took me a long time to figure out what was happening. It’s GitHub that does it, I believe, and it’s incredibly stubborn about it. The purge should do the trick for you. The cache will eventually get updated, but to see results right away, try the purge"

**Review by kcleland0818 — APPROVED** [review]

File: README.md (hunk 4 above)
Comment (linda-seiter) [reply README.md]: cool, thanks!

**Review by AlveeM — APPROVED** [review]

**Pull request merged**